### PR TITLE
feat: Confirm technical controls exist and are implemented

### LINKED ISSUE #310: Check for Technical Controls
**Extended Description**

As a FedRAMP reviewer, in order to understand the CSP's security posture with technically-focused controls, I want checks for the completion of specifically technically-focused controls in their system security plan.

**Preconditions**
 A review in approach on how to reuse the same method and unify messaging around overlapping checks for automation-focused controls. 18F#309
 
Refer to prop[@name='implementation-status'] in Section 5.3 of Guide_to_OSCAL-based_FedRAMP_System_Security_Plans.

**Acceptance Criteria**
- [x] Checks that flag that the following controls are not complete. 
  - [x] AC-02 (10), AC-03, AC-05, AC-06, AC-06 (01), AC-06 (05), AC-06 (07), AC-06 (09), AC-06 (10), AC-07,AC-07 (02), AC-08, AC-10, AC-11,  AC-11 (01), AC-17 (01), AC-17 (02), AC-18 (01), AC-19, AC-19 (5), AC-20, AC-20 (01), AC-20 (02), AC-21 
  - [x] AU-02 (03),  AU-05 (02), AU-06,  AU-06 (07),  AU-09 (02), AU-09 (04), AU-11
  - [x] CM-05, CM-05 (01), CM-10
  - [x] IA-04 (04), IA-05 (04), IA-06
  - [x] IR-02 (02), 
  - [x] SC-07 (05), SC-10"

**Story Tasks**
- [x] Tasks...

**Definition of Done**
 - [x] Acceptance criteria met
 - [x] Unit test coverage of our code > 95%  
 - [x] Automated code quality checks passed 
 - [x] Security reviewed and reported
 - [x] Reviewed against plain language guidelines
 - [x] Code must be self-documenting
 - [x] No local tech debt
 - [x] Load/performance tests passed – needs to be created/automated
 - [x] Documentation updated
 - [x] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
* add values to fedramp_values
* two assertions for ssp.sch
* pos & neg xspec tests for ssp.xspec for both assertions

close #310

### CODE REVIEW

File: src/validations/rules/ssp.sch [hunk 1]
```diff
@@ -3450,9 +3450,66 @@
...
+            <sch:let
+                name="missing-required-technical-controls"
+                value="$required-technical-controls[not(. = $implemented-controls)]" />
+            <sch:assert
+                diagnostics="technical-control-exists-diagnostic"
+                id="technical-control-exists"
+                role="error"
+                test="
+                    every $item in $missing-required-technical-controls
+                        satisfies exists(//oscal:implemented-requirement[@control-id eq $item])">Every required technical control is
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:3482]: The bind variable is wrong. It should have been `$required-technical-controls`. As is, it is trying to find missing+required technical controls which are missing and thus will not be found.

The assertion test can be simplified to `count($missing-required-technical-controls) eq 0` or an equivalent.

File: src/validations/rules/ssp.sch [hunk 2]
```diff
@@ -3450,9 +3450,66 @@
...
+                    then
+                        (
+                        if (exists(oscal:prop[@ns eq 'https://fedramp.gov/ns/oscal' and @name = 'implementation-status' and @value eq 'implemented']))
+                        then
+                            (true())
+                        else
+                            (false()))
+                    else
+                        (true())">The technical control-implementation <xsl:value-of
+                    select="@control-id" /> has a status of 'implemented'.</sch:assert>
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:3512]: To be precise this should be "has an **implementation-status** of 'implemented'.".

The context is not "technical control-implementation" - it is a `implemented-requirement` and the assertion is testing that the technical control's `implemented-requirement` implementation status is 'implemented') The the assertion message should be something like "Every technical control is fully implemented." or "Technical control xx-x is implemented".

File: src/validations/rules/ssp.sch [hunk 3]
```diff
@@ -5337,6 +5394,16 @@
             doc:assertion="import-profile-has-href-attribute"
             doc:context="oscal:import-profile"
             id="import-profile-has-href-attribute-diagnostic">The import-profile element lacks an href attribute.</sch:diagnostic>
+        <sch:diagnostic
+            doc:assertion="technical-control-exists"
+            doc:context="oscal:control-implementation"
+            id="technical-control-exists-diagnostic">The technical control implementation <sch:value-of
+                select="@control-id" /> does not exist in the SSP document.</sch:diagnostic>
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:5401]: The context of the associated assertion is `oscal:control-implementation`. Why is `@control-id` being referenced?

File: src/validations/rules/ssp.sch [hunk 4]
```diff
@@ -3450,9 +3450,66 @@
...
+            <sch:let
+                name="technical-controls"
+                value="$fedramp-values//fedramp:value-set[@name eq 'technical-control-id']//fedramp:enum/@value ! xs:string(.)" />
+            <sch:let
+                name="required-technical-controls"
+                value="$technical-controls[. = $required-controls]" />
+            <sch:let
+                name="missing-required-technical-controls"
+                value="$required-technical-controls[not(. = $implemented-controls)]" />
+            <sch:assert
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:3476]: Missing ` fedramp:specific="true"`.

File: src/validations/rules/ssp.sch [hunk 5]
```diff
@@ -3450,9 +3450,66 @@
...
+            <sch:let
+                name="sensitivity-level"
+                value="/ => lv:sensitivity-level()" />
+            <sch:let
+                name="selected-profile"
+                value="$sensitivity-level => lv:profile()" />
+            <sch:let
+                name="technical-controls"
+                value="$fedramp-values//fedramp:value-set[@name eq 'technical-control-id']//fedramp:enum/@value ! xs:string(.)" />
+            <sch:assert
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:3497]: Missing ` fedramp:specific="true"`.

File: src/validations/rules/ssp.sch (hunk 1 above)
Comment (markXLIX) [reply src/validations/rules/ssp.sch]: fixed

File: src/validations/rules/ssp.sch (hunk 2 above)
Comment (markXLIX) [reply src/validations/rules/ssp.sch]: fixed

File: src/validations/rules/ssp.sch (hunk 3 above)
Comment (markXLIX) [reply src/validations/rules/ssp.sch]: fixed

File: src/validations/rules/ssp.sch (hunk 5 above)
Comment (markXLIX) [reply src/validations/rules/ssp.sch:3496]: fixed

File: src/validations/rules/ssp.sch (hunk 4 above)
Comment (markXLIX) [reply src/validations/rules/ssp.sch:3476]: fixed

File: src/validations/rules/ssp.sch (hunk 1 above)
Comment (markXLIX) [reply src/validations/rules/ssp.sch]: I fixed this incorrectly into the last commit.  This is fixed (3rd commit) and working locally now.

**Pull request merged**